test/unit/resources/aws_db_parameter_group_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'helper'
+require 'aws_db_parameter_group'
+require 'aws-sdk-core'
+
+class AwsDbParameterGroupConstructorTest < Minitest::Test
+
+  def test_empty_params_not_ok
+    assert_raises(ArgumentError) { AwsDbParameterGroup.new(client_args: { stub_responses: true }) }
+  end
+
+  def test_empty_param_arg_not_ok
+    assert_raises(ArgumentError) { AwsDbParameterGroup.new(db_parameter_group_name: '', client_args: { stub_responses: true }) }
+  end
+
+  def test_rejects_unrecognized_params
+    assert_raises(ArgumentError) { AwsDbParameterGroup.new(unexpected: 9) }
+  end
+
+  def test_parameter_non_existing
+    refute AwsDbParameterGroup.new(db_parameter_group_name: 'test-1', client_args: { stub_responses: true }).exists?
+  end
+end
+
+class AwsDbParameterGroupSuccessPathTest < Minitest::Test
+
+  def setup
+    data = {}
+    data[:method] = :describe_db_parameter_groups
+    mock_parameter = {}
```

Comment:
This approach is absolutely fine, but you may consider using mocks to follow DRY..